.github/workflows/reports.yml

```diff
@@ -262,8 +261,143 @@ jobs:
         working-directory: ./test_programs
         run: |
           ./compilation_report.sh
+          cat compilation_report.json
           mv compilation_report.json ../compilation_report.json
 
+      - name: Upload compilation report
+        uses: actions/upload-artifact@v4
+        with:
+          name: in_progress_compilation_report
+          path: compilation_report.json
+          retention-days: 3
+          overwrite: true
+
+  external_repo_compilation_report:
+    needs: [generate_compilation_report]
```

Comment:
We should be able to relax this restriction (and then add `generate_compilation_report` as a new dependency of the report accumulation job)

Reply:
done